src/Transfer/Sources/Firebase.php

```diff
@@ -0,0 +1,522 @@
+                    $user["disabled"]
+                );
+            }
+
+            $this->callback($users);
+
+            if (count($result) < $batchSize) {
+                break;
+            }
+        }
+    }
+
+    private function calculateUserType(array $providerData): array
+    {
+        if (count($providerData) === 0) {
+            return [User::TYPE_ANONYMOUS];
+        }
+
+        $types = [];
+
+        foreach ($providerData as $provider) {
+            switch ($provider["providerId"]) {
+                case 'password':
+                    $types[] = User::TYPE_EMAIL;
+                    break;
+                case 'phone':
+                    $types[] = User::TYPE_PHONE;
+                    break;
+                default:
+                    $types[] = User::TYPE_OAUTH;
+                    break;
+            }
+        }
+
+        return $types;
+    }
+
+    public function exportDatabasesGroup(int $batchSize, array $resources)
+    {
+        if (in_array(Resource::TYPE_DATABASE, $resources)) {
+            $database = new Database('default', '(default)');
+            $this->callback([$database]);
+        }
+
+        if (in_array(Resource::TYPE_COLLECTION, $resources)) {
+            $this->handleDBData($batchSize, in_array(Resource::TYPE_DOCUMENT, $resources), $database);
+        }
+    }
+
+    private function handleDBData(int $batchSize, bool $pushDocuments, Database $database)
+    {
+        $baseURL = "https://firestore.googleapis.com/v1/{$this->projectID}/databases/(default)";
+
+        $nextPageToken = null;
+        $allCollections = [];
+        while (true) {
+            $collections = [];
+
+            $result = $this->call('POST', $baseURL . ':listCollectionIds', [
+                'Content-Type' => 'application/json',
+            ], [
+                'pageSize' => $batchSize,
+                'pageToken' => $nextPageToken
+            ]);
+
+            // Transfer Collections
+            foreach ($result['collectionIds'] as $collection) {
+                $collections[] = new Collection($database, $collection, $collection);
+            }
+
+            if (count($collections) !== 0) {
+                $allCollections = array_merge($allCollections, $collections);
+                $this->callback($collections);
+            } else {
+                return;
+            }
+
+            // Transfer Documents and Calculate Schema
+            foreach ($collections as $collection) {
+                $this->handleCollection($collection, $batchSize, $pushDocuments);
+            }
+
+            if (count($result['collectionIds']) < $batchSize) {
+                break;
+            }
+
+            $nextPageToken = $result['nextPageToken'] ?? null;
+        }
+    }
+
+    private function convertAttribute(Collection $collection, string $key, array $field): Attribute
+    {
+        if (array_key_exists("booleanValue", $field)) {
```

Comment:
Let's use a proper switch case.

Reply:
Not sure if we can do that as we are checking for the existence of a key and not checking if a value matches